src/core/createCookieTransfer.js

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { endsWith, includes, isNamespacedCookieName } from "../utils";
+import convertTimes, { DAY, SECOND } from "../utils/convertTimes";
+
+const STATE_STORE_HANDLE_TYPE = "state:store";
+
+export default ({ cookieJar, orgId, apexDomain }) => {
+  // For AMCV, we intentionally do not replace the @ sign that's
+  // found in the Org ID because it's never replaced when
+  // written or read by Visitor.js.
+  const legacyCookieNames = ["s_ecid", encodeURIComponent(`AMCV_${orgId}`)];
+  return {
+    /**
+     * When sending to a third-party endpoint, the endpoint won't be able to
+     * access first-party cookies, therefore we transfer cookies into
+     * the request body so they can be read by the server.
+     */
+    cookiesToPayload(payload, endpointDomain) {
+      const isEndpointFirstParty = endsWith(endpointDomain, apexDomain);
+
+      // If the endpoint is first-party, there's no need to transfer cookies
+      // to the payload since they'll be automatically passed through cookie
+      // headers.
+      if (isEndpointFirstParty) {
+        return;
+      }
+
+      const cookies = cookieJar.get();
+
+      const state = Object.keys(cookies)
+        .filter(name => {
+          // We have a contract with the server that we will pass
+          // all cookies whose names are namespaced according to the
+          // logic in isNamespacedCookieName as well as any legacy
+          // cookie names (so that the server can handle migrating
+          // identities on websites previously using Visitor.js)
+          return (
+            isNamespacedCookieName(orgId, name) ||
+            includes(legacyCookieNames, name)
+          );
+        })
+        .map(qualifyingCookieName => {
+          return {
+            key: qualifyingCookieName,
+            value: cookies[qualifyingCookieName]
+          };
+        });
+
+      if (state.length) {
+        payload.mergeMeta({
+          state
+        });
+      }
+    },
+    /**
+     * When receiving from a third-party endpoint, the endpoint won't be able to
+     * write first-party cookies, therefore we write first-party cookies
+     * as directed in the response body.
+     */
+    responseToCookies(response) {
+      response.getPayloadsByType(STATE_STORE_HANDLE_TYPE).forEach(stateItem => {
+        const options = { domain: apexDomain };
```

Comment:
The only slightly risky assumption we are making here, is that the server and our apex detection will return the same value :)

Reply:
It's true. I'm not sure how risky that it, but maybe it is best to send Konductor the apex domain.

Reply:
Let' bring that up with the Konductor folks on our next meeting. Taking notes.